optimum/habana/transformers/generation/utils.py

```diff
@@ -1089,10 +1093,18 @@ def generate(
                 calculated_max_length = input_ids.shape[-1] + generation_config.max_new_tokens + num_virtual_tokens
             if generation_config.use_cache and generation_config.reuse_cache:
                 bs, _ = input_ids.shape
+                cache_device = "cpu" if generation_config.kv_cache_on_host else "hpu"
                 if not is_greedy_or_beam_and_bucket:
-                    unwrap_deepspeed_model(self).allocate_kv_cache(
-                        bs * generation_config.num_beams, calculated_max_length, token_idx + num_virtual_tokens
+                    if generation_config.kv_cache_on_host and self.config.model_type in ["llama"]:
+                        print("Allocate KV Cache on CPU...")
+                        unwrap_deepspeed_model(self).allocate_kv_cache(
+                        bs * generation_config.num_beams, calculated_max_length, token_idx + num_virtual_tokens,
+                        device=cache_device
                     )
+                    else:
+                        unwrap_deepspeed_model(self).allocate_kv_cache(
+                            bs * generation_config.num_beams, calculated_max_length, token_idx + num_virtual_tokens
+                        )
```

Comment:
Thanks, I have updated it in [74e94ff](https://github.com/huggingface/optimum-habana/pull/1182/commits/74e94ff520fefcf0b5109bf9a7d5acfec6f1347d). However, I can not remove [the else line](https://github.com/zhentaoyu/optimum-habana/blob/cpu_sdpa/optimum/habana/transformers/generation/utils.py#L1104) because I only modified the `modeling_llama.py` for this experimental feature.